readthedocs/config/config.py

```diff
@@ -823,12 +823,16 @@ def validate_final_doc_type(self):
         should support per version doctype, but we need to
         adapt the rtd code for that.
         """
-        if self.doctype != self.defaults.get('doctype', 'sphinx'):
+        dashboard_doctype = self.defaults.get('doctype', 'sphinx')
+        if self.doctype != dashboard_doctype:
             key = 'mkdocs' if self.doctype == 'mkdocs' else 'sphinx'
             self.error(
                 key,
-                'Your documentation type should match '
-                'the one from the admin panel of your project.',
+                'Your project is configured as "{doctype}" in your admin '
+                'dashboard, but there was a "{key}" key specified.'.format(
```

Comment:
One last nitpick. The error is sort of inverted i feel -- we don't actually tell the user the they need a "sphinx" key, just that they can't have a "mkdocs" key. I feel we need to be clearer here, ie:

```
Your project is configured as "Sphinx" in your project admin dashboard, but there is no "sphinx" key specified.
```

Also, for some of the odd documentation types, it looks like we're putting the slug in there, so:

```
Your project is configured as "sphinx_htmldir" ...
```

I feel like we want the verbose name instead. This shouldn't be a huge change, just using the model field to resolve the verbose name, but if it is a large change, it's not horribly important. This piece is temporary and it's probably close enough for most users.

Reply:
Mmm, I'm not sure about accessing the db from the config module (we don't do that there). I'm changing the error message.